zerver/lib/emoji.py

```diff
@@ -111,13 +111,13 @@ def check_remove_custom_emoji(user_profile: UserProfile, emoji_name: str) -> Non
 
 def check_valid_emoji_name(emoji_name: str) -> None:
     if emoji_name:
-        if re.match(r"^[0-9a-z.\-_]+(?<![.\-_])$", emoji_name):
+        if re.match(r"^[0-9a-z\-_]+(?<![\-_])$", emoji_name):
             return
-        if re.match(r"^[0-9a-z.\-_]+$", emoji_name):
+        if re.match(r"^[0-9a-z\-_]+$", emoji_name):
             raise JsonableError(_("Emoji names must end with either a letter or number."))
         raise JsonableError(
             _(
-                "Emoji names must contain only numbers, lowercase English letters, spaces, dashes, underscores, and periods.",
+                "Emoji names must contain only numbers, lowercase English letters, spaces, dashes, and underscores.",
```

Comment:
Technically speaking, `0-9` are [digits](https://en.wikipedia.org/wiki/Numerical_digit), not [numbers](https://en.wikipedia.org/wiki/Number).
```suggestion
                "Emoji names must contain only digits, lowercase English letters, spaces, dashes, and underscores.",
```

Reply:
Yeah I agree! changed it now 👍